### PR TITLE
test: Define and adjust migration tests to verify external id is populated for new API compatible schemas

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-260661

Add migration tests to ensure correct transition from old state that does not have `external_id` defined to new state which populates `external_id` as it is invoking the new API in the read. This applies to all schemas which are compatible with the new API (all `num_shards` = 1).

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by AgustinBettati** [comment]
failing test `TestAccClusterAdvancedCluster_singleShardedMultiCloud` related to CLOUDP-258714

**Review by EspenAlbert — APPROVED** [review]
Nice!

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go [hunk 1]
```diff
@@ -47,7 +52,89 @@ func TestMigAdvancedCluster_replicaSetMultiCloud(t *testing.T) {
...
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				Config:                   config,
+				Check:                    checkGeoShardedOldSchema(clusterName, 1, 1, true, true), // external_id will be present in latest version
+			},
+		},
+	})
+}
+
+func TestMigAdvancedCluster_symmetricGeoShardedOldSchema(t *testing.T) {
+	acc.SkipTestForCI(t) // TODO: CLOUDP-260154 for ensuring old schema is supported
```
Comment (lantoli) [inline internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:122]: is this comment still valid?

**Review by lantoli — APPROVED** [review]

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:122]: yes still valid, the test is currently failing and we have a fix pending from API side.

**Pull request merged**